libgit/autogit_node_wrappers.go

```diff
@@ -82,12 +86,19 @@ var _ libkbfs.Node = (*repoNode)(nil)
 
 func newRepoNode(n libkbfs.Node, am *AutogitManager, h *libkbfs.TlfHandle,
 	repoName string) *repoNode {
-	return &repoNode{
+	rn := &repoNode{
 		Node:     n,
 		am:       am,
 		h:        h,
 		repoName: repoName,
 	}
+	// We can't rely on a particular repo node being passed back into
+	// libkbfs by callers, since they may not keep a reference to it
+	// after looking it up, and `WrapChild` makes a new `repoNode` for
+	// each call to it, even for the same underlying NodeID.  So we
+	// keep the populated state in the AutogitManager.
```

Comment:
What about if the user's client hasn't yet created a `folderBranchOps` for that TLF?

Reply:
Which TLF are you asking about, the source git repo TLF or the destination checkout TLF?  The `repoNode` here is from the destination TLF, and will only exist if the callbacks have been triggered by the corresponding `folderBranchOps` instance.

If you're talking about the source repo TLF, that's orthogonal to this `isRepoNodePopulated` call, but in general it will get created by the autogit code as soon as it tries to read the repo to service the `Reset` call. No user action is needed.

Reply:
I'm just trying to understand the flow of what happens to an auto-checkout in dstTlf when srcTlf/fooRepo receives a push. And which client does the work.

Reply:
When a client instance first reads the autogit dir in the dest, it necessarily ends up reading from the source repo TLF.  When it does that, it registers for updates with that TLF's `folderBranchOps`, and handles the updates subsequently.  (If multiple client instances have read it, the reset will be done by whoever first takes the server-side lock.)

The client that wrote to the repo doesn't do anything in terms of updating the autogit repo.

Reply:
OK so if no clients have a `folderBranchOps` on the src (and thus have registered for updates), the auto-checkout doesn't get updated on a push. So then when the next read attempt happens, all that scaffolding gets set up, the client does a `Pull`, and also registers for updates.

Does that sound right?

Reply:
Yep, exactly.